lib/berkeley_library/docker/railtie.rb

```diff
@@ -2,10 +2,28 @@
 
 module BerkeleyLibrary
   module Docker
-    class Railtie < Rails::Railtie
-      NAME = 'berkeley_library-docker.load_secrets'.freeze
+    class Railtie < ::Rails::Railtie
+      include Logging
 
-      initializer(NAME, after: :initialize_logger) { Secret.load_secrets! }
+      def secrets
+        @secrets ||= {}
```

Comment:
Since we're deferring logging (on account of Rails initializing the logger after application.rb), we needed a way to track which secrets were actually loaded by the railtie. I chose to keep that state logic in the railtie rather than the gem, that way conventional non-Rails consumers can have a choice over how (if at all) secret loading is logged.